src/main/resources/templates/index.html

```diff
@@ -3,59 +3,18 @@
 <head>
     <meta charset="UTF-8">
     <title>Market Data</title>
-    <style>
-        body{font-family: Consolas,serif;}
-        .panel{ width: 400px; height: 400px; border: solid black 1px}
-    </style>
+    <link rel="stylesheet" href="style.css">
+    <script src="script.js"></script>
 </head>
+<body>
 <h1>Market Data API</h1>
-<h2 id="symbol"></h2>
-<body >
+<h2 id="symbol"> </h2> <!-- I had a misunderstanding about this. I see that your JS is using innerHTML to work with this element. This is fine. It was throwing an error before because your JavaScript was not being called upon as it was located in the wrong place.-->
+<img src="placeholder.png" alt ="Placeholder image of a chart"> <!--If you want to add an image, do it like this using an alt tag. Using empty headers is bad for accessibility.-->
     <h3>Quotes</h3>
     <div id="quotes" class="panel"></div>
     <h3>Trades</h3>
     <div id="trades" class="panel"></div>
     <h3>Bars</h3>
     <div id="bars" class="panel"></div>
 </body>
-<script>
-    document.getElementById("symbol").innerHTML = window.location.href;
-    urlSymbol = window.location.href;
-    tickerSymbol = urlSymbol.substring(urlSymbol.lastIndexOf("/") + 1);
-    document.getElementById("symbol").innerHTML = tickerSymbol;
-
-    const url = "wss://stream.data.alpaca.markets/v2/iex"
-    const auth = {"action":"auth","key":"AKFOO6Z1ZA5J9L4WR4XD","secret":"NGCMczuSOV3LSdOE7v4azBhP1fwzFNyPHHVCH4rS"}
-    const subscribe = {"action":"subscribe","trades":[tickerSymbol],"quotes":[tickerSymbol],"bars":[tickerSymbol]}
-
-    const socket = new WebSocket(url);
-    socket.onmessage = function(event){
-        const data = JSON.parse(event.data)
-        const message = data[0]['msg']
-
-        if (message === 'connected') {
-            socket.send(JSON.stringify(auth))
-        }
-
-        if (message === 'authenticated') {
-            socket.send(JSON.stringify(subscribe))
-        }
-
-        for (const key in data) {
-            console.log(data[key])
-            if(data[key]["T"] === 't'){
-                console.log("trade")
-                console.log(data[key])
-            }
-            if(data[key]["T"] === 'q'){
-                console.log("quote")
-                console.log(data[key])
-            }
-            if(data[key]["T"] === 'b'){
-                console.log("bar")
-                console.log(data[key])
-            }
-        }
-    }
-</script>
```

Comment:
There were a few issues with this. When first looking at index.html, this was located outside of the header and body tags. Therefore, it wasn't functioning whatsoever. It was also clogging up index.html. I went ahead and linked it in the header using the script tag and moved it to an external JS file called script.js.

Reply:
Really good.